app/models/edm/web_resource.rb

```diff
@@ -119,6 +121,14 @@ def rdf_about
       media&.url
     end
 
+    def remove_versions
+      media.versions.each_key { |key| media.send(key).remove! }
+    end
+
+    def set_deleted_resource
```

Comment:
Could we extract this into a model concern? Something like:
```ruby
module RecordableDeletion
  extend ActiveSupport::Concern

  included do
    before_destroy :create_deleted_resource
  end

  def create_deleted_resource # renamed from set_deleted_resource
    # ...
  end
end
```

We should then also include this in `Contribution`, for the storage of `deleted_at` and `deleted_by` which are not preserved on the wiped contribution documents.